install-deps.sh

```diff
@@ -146,19 +146,49 @@ function install_pkg_on_ubuntu {
     fi
 }
 
+boost_ver=1.82
+
+function clean_boost_on_ubuntu {
+    ci_debug "Running clean_boost_on_ubuntu() in install-deps.sh"
+    # Find currently installed version. This may not work if there are
+    # multiple versions installed.
+    local installed_ver=$(apt -qq list --installed ceph-libboost*-dev 2>/dev/null |
+                              grep -e 'libboost[0-9].[0-9]\+-dev' |
+                              cut -d' ' -f2 |
+                              cut -d'.' -f1,2)
+    # If installed_ver contains a space, we can't really count on it,
+    # but otherwise, bail out if the version installed is the version
+    # we want.
+    if test -n "$installed_ver" && echo "$installed_ver" | grep -v -q ' '; then
+        if echo "$installed_ver" | grep -q "^$boost_ver"; then
+            return
+        fi
+    fi
+
+    # Historical packages
+    $SUDO rm -f "/etc/apt/sources.list.d/ceph-libboost*.list"
+    # Currently used
+    $SUDO rm -f "/etc/apt/sources.list.d/libboost.list"
+    # Refresh package list so things aren't in the available list.
+    $SUDO apt-get update -y
+    # Remove all ceph-libboost dev packages except, possibly, those of
+    # the version we currently want.
+    apt -qq list --installed 'ceph-libboost*-dev' 2> /dev/null | \
+	cut -f 1 -d / | \
+	xargs $SUDO apt-get -y remove
```

Comment:
With the filter removed, do we still need `cut` and `xargs` or would the current `$SUDO env DEBIAN_FRONTEND=noninteractive apt-get -y remove "ceph-libboost.*${installed_ver}.*"` (note wildcards) do the job?

Reply:
We don't want `${installed_ver}`. But I think `ceph-libboost.*` should be fine.

I had originally intended to keep it as `-dev` as the minimal conservative operation that would unbreak the build, but on consideration I don't think there's any reason to spare the non-dev packages after we're removing the repos they came from, and having them doesn't buy us anything.